content/docs/user-guide/assets/pipeline/asset-dependencies-and-identifiers.md

```diff
@@ -7,16 +7,22 @@ toc: true
 ---
 
 Both source and product assets might have dependencies. Source asset and job dependencies are inputs to the **Asset Pipeline** and asset processing jobs. Product dependencies are outputs from the Asset Pipeline and are used to load and package product assets. Asset identifiers help ensure both types of asset dependencies are met.
+
+## Source dependencies
+
+Source dependencies are a process time concept.
+
+A source dependency triggers **Asset Processor** to rerun the process job whenever the file declared as a source dependency changes. For example, a `.assetinfo` file is a source dependency and triggers a process job on its source asset when it is modified.
 
-## Source and job dependencies
+## Job dependencies
 
-Source and job dependencies are a process time concept.
+Job dependencies are a process time concept.
 
-A source dependency triggers **Asset Processor** to rerun the process job whenever a source asset changes. For example, a `.assetinfo` file is a source dependency and triggers a process job on its source asset when it is modified.
-
-A job dependency requires the product asset of another process job before this process job can be run. The shader and material relationship is an example of a job dependency. The material cannot be processed until the shader product asset is generated because the material requires information about the shader product asset.
+A job dependency is declared against another job. If a job matching that description exists in the job queue, when the job with the dependency is up next to run, it will first make sure all of its job dependencies are run. Also, if a job dependency runs again, it will trigger any jobs with that dependency to run. The shader and material relationship is an example of a job dependency. The material cannot be processed until the shader job is completed because the material requires information from product assets of the shader job. If the source shader asset is changed, causing the shader job to re-run, then the material job with that job dependency will re-run after the shader job finishes.

```

Comment:
"If a job dependency runs again" is kind of confusing.  Assuming I'm understanding correctly, maybe something like "When a job runs again, it will trigger any jobs that depend on it to run".

Also I would call out the need for reading a compiled asset as the key reason for using a job dependency, its sort of mentioned off hand here as an example but I think its a really important distinction for people to understand.

Reply:
Updated to use your suggested phrasing. I also added a note near the beginning to make the intent specifically called out.